index.js

```diff
@@ -23,74 +21,110 @@ const rp = require('request-promise');
  * 	- attach(params, file)
  *
  */
+
 const Servicetrade = (options) => {
+    options = options || {};
+    options.baseUrl = options.baseUrl || 'https://api.servicetrade.com';
+
+    let request = axios.create({
+        baseURL: options.baseUrl + '/api'
+    });
+
+    if (options.cookie) {
+        request.defaults.headers.Cookie = options.cookie;
+    }
+
+    if (!options.disableRefreshAuth) {
+        const refreshAuthLogic = async function(failedRequest) {
+            request.defaults.headers.Cookie = null;
+
+            let auth = {
+                username: options.username,
+                password: options.password
+            };
+            try {
+                await request.post('/auth', auth);
+            } catch (e) {
+                request.defaults.headers.Cookie = null;
+                throw err;
+            }
+        };
+        createAuthRefreshInterceptor.default(request, refreshAuthLogic);
+    }
+
+    request.interceptors.response.use(function(response) {
+        if (
+            !request.defaults.headers.Cookie ||
+            !Object.keys(request.defaults.headers.Cookie).length
+        ) {
+            if (response.headers && response.headers['set-cookie']) {
+                const [cookie] = response.headers['set-cookie'];
+                request.defaults.headers.Cookie = cookie;
+            }
+        }
+
+        // detect if it response which we have after refresh token
+        if (!response.config && !response.headers && !response.request) {
+            return response;
+        }
+        return response && response.data && response.data.data ? response.data.data : null;
+    });
+
+    return {
+        getRequestObject: () => {
+          return request;
+        },
+        setRequestObject: (req) => {
+          request = req;
+        },
+
```

Comment:
do `getRequestObject` and `setRequestObject` need to be exposed in `exports`?  There weren't any analogous functions in the old code, and they don't have any test coverage.